Objectstore flaky test in TestFlushWorkersCancelledContext

### Description

```
--- FAIL: TestWorkerSuite (0.02s)
    --- FAIL: TestWorkerSuite/TestFlushWorkersCancelledContext (0.00s)
        /home/jenkins/go/src/github.com/juju/juju/internal/worker/objectstore/worker_test.go:226
        worker_test.go:238: 
                c.Assert(err, tc.ErrorIs, context.Canceled)
            ... obtained = nil
            ... error *errors.errorString = &errors.errorString{s:"context canceled"} ("context canceled")
        runner.go:314: DEBUG: killing runner "object-store"
        runner.go:436: INFO: runner "object-store" is dying
FAIL
FAIL	github.com/juju/juju/internal/worker/objectstore	0.101s
```

### Juju version

4.1-HEAD

### Cloud

_No response_

### Expected behaviour

Tests pass

### Reproduce / Test

Run TestFlushWorkersCancelledContext

### Notes & References

_No response_

--- a/internal/worker/objectstore/worker.go
+++ b/internal/worker/objectstore/worker.go
@@ -331,6 +331,11 @@ func (w *objectStoreWorker) GetObjectStore(ctx context.Context, namespace string
 // the runner. This is only called from the main loop goroutine, so no
 // concurrent worker additions can occur during iteration.
 func (w *objectStoreWorker) stopAndRemoveAllWorkers(ctx context.Context) error {
+	// If the context is already done, return early with the context error.
+	if err := ctx.Err(); err != nil {
+		return errors.Trace(err)
+	}
+
 	for _, namespace := range w.runner.WorkerNames() {
 		err := w.runner.StopAndRemoveWorker(namespace, ctx.Done())
 		if err != nil && !errors.Is(err, errors.NotFound) {

--- a/internal/worker/objectstore/worker_test.go
+++ b/internal/worker/objectstore/worker_test.go
@@ -227,7 +227,7 @@ func (s *workerSuite) TestFlushWorkersCancelledContext(c *tc.C) {
 	defer s.setupMocks(c).Finish()
 
 	w := s.newWorker(c)
-	defer workertest.CleanKill(c, w)
+	defer workertest.DirtyKill(c, w)
 
 	s.ensureStartup(c)
 
@@ -236,8 +236,6 @@ func (s *workerSuite) TestFlushWorkersCancelledContext(c *tc.C) {
 
 	err := w.FlushWorkers(ctx)
 	c.Assert(err, tc.ErrorIs, context.Canceled)
-
-	workertest.CleanKill(c, w)
 }
 
 func (s *workerSuite) newWorker(c *tc.C) *objectStoreWorker {

